web/impact/impact/v1/views/cancel_office_hour_reservation_view.py

```diff
@@ -0,0 +1,94 @@
+from pytz import timezone
+
+from django.contrib.auth import get_user_model
+from django.template import loader
+
+from rest_framework.response import Response
+
+from impact.minimal_email_handler import MinimalEmailHandler as email_handler
+from impact.v1.views.impact_view import ImpactView
+from impact.permissions.v1_api_permissions import (
+    OfficeHourFinalistPermission,
+)
+from accelerator_abstract.models.base_user_utils import is_employee
+from accelerator.models import MentorProgramOfficeHour
+
+User = get_user_model()
+
+mentor_template_name = "cancel_office_hour_reservation_email_to_mentor.html"
+finalist_template_name = ("cancel_office_hour_reservation_email_to_finalist."
+                          "html")
+SUBJECT_LINE = "MassChallenge | Cancelled Office Hours with {} {}"
+
+
+class CancelOfficeHourReservationView(ImpactView):
+    permission_classes = (OfficeHourFinalistPermission, )
+    view_name = "cancel_office_hours_reservation"
+    posted_fields = ["office_hour_id",
+                     "message"]
+
+    def _extract_field(self, data, field):
+        setattr(self, field, data.get(field, None))
+
+    def _extract_posted_data(self, data, fields):
+        for field in fields:
+            self._extract_field(data, field)
+
+    def post(self, request):
+        self._extract_posted_data(request.data, self.posted_fields)
+        self.user = request.user
+        self.office_hour = MentorProgramOfficeHour.objects.get(
+            pk=self.office_hour_id)
+        success = False
```

Comment:
If the cancellation fails, return a reason ("You do not have permission to cancel that reservation" or "That reservation does not exist")